local_server/local-server-generate-elevation-profile/src/main/java/com/esri/samples/local_server_generate_elevation_profile/LocalServerGenerateElevationProfileController.java

```diff
@@ -0,0 +1,569 @@
+            e.printStackTrace();
+          }
+
+          localGPService.addStatusChangedListener(s -> {
+            // create geoprocessing task once local geoprocessing service is started 
+            if (s.getNewStatus() == LocalServerStatus.STARTED) {
+              // add the name of the model used to create the gpkx in ArcGIS Pro to the Url of the local 
+              // geoprocessing task
+              // e.g. the model name in this sample's gpkx created in ArcGIS Pro is CreateElevationProfileModel
+              gpTask = new GeoprocessingTask(localGPService.getUrl() + "/CreateElevationProfileModel");
+
+              // handle UI behaviour
+              instructionsLabel.setText("Draw a polyline on the scene with the 'Draw Polyline' button");
+              drawPolylineButton.setDisable(false);
+              progressBar.setVisible(false);
+            }
+          });
+          localGPService.startAsync();
+          
+        } else if (server.getStatus() == LocalServerStatus.FAILED) {
+          // display an information alert and close the application if the server status failed to start
+          showMessage("Local Server Status Error", "Local Server Status Failed to start.");
+        }
+      });
+      server.startAsync();
+      
+    } else {
+      // display an information alert and close the application if a local server install path couldn't be located
+      showMessage("Local Server Install Error", "Local Server install path couldn't be located.");
+    }
+  }
+
+  /**
+   * Generates an elevation profile when the "Generate Elevation Profile" button is clicked. The elevation profile is
+   * generated from an input raster (Isle of Arran Lidar data) and a polyline sketched by the user.
+   */
+  @FXML
+  protected void handleGenerateElevationProfile() {
+
+    generateProfileButton.setDisable(true);
+    // tracking progress of generating elevation profile
+    progressBar.setVisible(true);
+
+    // create the feature collection table from sketched polyline
+    createFeatureCollectionTableWithPolylineFeature();
+
+    // create default parameters and get their inputs
+    var defaultParamsListener = gpTask.createDefaultParametersAsync();
+    defaultParamsListener.addDoneListener(() -> {
+      try {
+        var params = defaultParamsListener.get();
+        var inputParams = params.getInputs();
+
+        // input polyline path
+        // name of input parameter, input type (geoprocessing feature, pointing to polyline)
+        inputParams.put("Input_Polyline", new GeoprocessingFeatures(featureCollection.getTables().get(0)));
+
+        // input raster data
+        // name of input parameter, input type (geoprocessing string pointing to raster file)
+        inputParams.put("Input_Raster", new GeoprocessingString(arranRaster.getPath()));
+
+        // create geoprocessing job from the geoprocessing parameters to show elevation profile on the scene
+        GeoprocessingJob gpJob = gpTask.createJob(params);
```

Comment:
var's are a curious thing! they are good to use when doing a standard instantiation where you are repeated the class name 3 or 4 times. but in an example like this where it's not clear from the method (createJob) what type of job it's creating, it's better to keep the class declared rather than use var for readability.